AudioSinkExample/AudioSinks/ExampleSpeechRecognizer.m

```diff
@@ -0,0 +1,142 @@
+
+@end
+
+@implementation ExampleSpeechRecognizer
+
+- (instancetype)initWithAudioTrack:(TVIAudioTrack *)audioTrack
+                        identifier:(NSString *)identifier
+                     resultHandler:(void (^)(SFSpeechRecognitionResult * result, NSError * error))resultHandler {
+    self = [super init];
+
+    if (self != nil) {
+        _speechRecognizer = [[SFSpeechRecognizer alloc] init];
+        _speechRecognizer.defaultTaskHint = SFSpeechRecognitionTaskHintDictation;
+
+        _speechRequest = [[SFSpeechAudioBufferRecognitionRequest alloc] init];
+        _speechRequest.shouldReportPartialResults = YES;
+
+        // The iOS audio device captures in mono.
+        // The mixer produces stereo audio for each remote Participant, even if they send mono audio.
+        _numberOfChannels = [audioTrack isKindOfClass:[TVILocalAudioTrack class]] ? 1 : 2;
+
+        __weak typeof(self) weakSelf = self;
+        _speechTask = [_speechRecognizer recognitionTaskWithRequest:_speechRequest resultHandler:^(SFSpeechRecognitionResult * _Nullable result, NSError * _Nullable error) {
+            __strong typeof(self) strongSelf = weakSelf;
+            if (result) {
+                strongSelf.speechResult = result.bestTranscription.formattedString;
+            } else {
+                // TODO: CE - do we handle errors or let our owner?
+                NSLog(@"Speech recognition error: %@", error);
+            }
+
+            resultHandler(result, error);
+        }];
+
+        _audioTrack = audioTrack;
+        [_audioTrack addSink:self];
+        _identifier = identifier;
+    }
+
+    return self;
+}
+
+- (void)dealloc {
+    [self.speechTask cancel];
+}
+
+- (void)stopRecognizing {
+    [self.audioTrack removeSink:self];
+
+    [self.speechTask finish];
+    self.speechRequest = nil;
+    self.speechRecognizer = nil;
+
+    if (self.speechConverter != NULL) {
+        AudioConverterDispose(self.speechConverter);
+        self.speechConverter = NULL;
+    }
+}
+
+#pragma mark - TVIAudioSink
+
+- (void)renderSample:(CMSampleBufferRef)audioSample {
+    CMAudioFormatDescriptionRef coreMediaFormat = (CMAudioFormatDescriptionRef)CMSampleBufferGetFormatDescription(audioSample);
+    const AudioStreamBasicDescription *basicDescription = CMAudioFormatDescriptionGetStreamBasicDescription(coreMediaFormat);
+
+    // Detect and discard the initial invalid samples...
+    // Waits for the track to start producing the expected audio channels.
+    if (basicDescription->mChannelsPerFrame != _numberOfChannels) {
+        return;
+    }
+
+    AVAudioFrameCount frameCount = (AVAudioFrameCount)CMSampleBufferGetNumSamples(audioSample);
+    AVAudioFormat *avAudioFormat = [[AVAudioFormat alloc] initWithCommonFormat:AVAudioPCMFormatInt16
+                                                                    sampleRate:basicDescription->mSampleRate
+                                                                      channels:1
```

Comment:
We are using 1 and 2 constant at other places too. use const instead for readability ?  e.g `kChannelsCountMono` and `kChannnelsCountStereo`

Reply:
This is pretty minor - but I can address it.